Model intermediate table

cantordust opened the pull request:
Intermediate table structure for filtering segmentations by model. This may evolve to incorporate the subset definition.

Peter9192 committed:
fix bbox overshooting by one cell

cantordust committed:
- WIP: Intermediate table for image-model lookup.

cantordust committed:
- Updated the database structure for the Maskformer segmentation pipeline

cantordust committed:
- Updated the segment_images endpoint

cantordust committed:
- Moved functions from the Maskformer segmentation pipeline to the `Project` class.

cantordust committed:
- Moved the `flatten_labels()` function (to be deprecated soon) intto `utils/functions.py`.
- Deprecating the `models.base.ModelBase` class.

cantordust committed:
- Updated the MaskFormer segmentation flow, image-model M2M table now works as expected (but might be extended with extra columns).

Peter9192 commented on src/streetscapes/models/maskformer/functions.py:
```diff
@@ -141,23 +105,27 @@ def segment_images(
-    to_process, to_overwrite = get_unprocessed_images(project, image_paths, overwrite)
+    (processed, unprocessed) = project.get_image_status(image_paths, "maskformer")
+    if overwrite:
+        processed = {}
```
Don't you need to add the processed images to `unprocessed` then?

cantordust replied:
Yeah, that was a leftover from a previous attempt at filtering the processed images. Fixed now - processing the same image again would result in the same UUID being used for the segmentation.

cantordust committed:
- Fixed a bug with the M2M table (UUID of existing segmentations not picked up correctly).
- Small updates in the `MaskFormer` segmentation workflow

cantordust committed:
- Updated `uv.lock`

cantordust committed:
- Updated model response schema & segmentation saving format for `MaskFormer`. Saving the segmentation files is very slow...:(

cantordust committed:
- Updated mask representation to use masked arrays. Slightly faster, but not by much.

TODO
====
- Find a better representation for the masks.

cantordust committed:
- Restored segmentation masks to the original format (not binary but with instance IDs)

Peter9192 committed:
Implement basic structure of BMFS like maskformer, but hit circular import

Peter9192 committed:
BFMS now works up to saving the result

Peter9192 committed:
Update src/streetscapes/models/bfms/model.py

Co-authored-by: Alexander Hadjiivanov <43831101+cantordust@users.noreply.github.com>

cantordust committed:
Merge pull request #87 from Urban-M4/bfms

Add BFMS model using ray serve

cantordust committed:
- Updated the BFMS model CLI, service and database.
- Model parameters are now stored as a binary string serialised using `orjson.dumps()` (this is also the case for MaskFormer). This is because DuckDB treats the JSON field as a struct that cannot be empty, so passing an empty dictionary triggers an error. Please deserialise the parameters with `orjson.loads()`.
- Using `uuid7()` from `uuid7gen` instead of the built-in `uuid` module until the minimal Python version is bumped to >=3.14.

cantordust committed:
- Updated the CLI and database schema for MaskFormer.
- Moved the CLI function into its owm module (`models/maskformer/cli.py`; not to be confused with Clippy :) ). Also merged the `segment_images` function into `cli.py`.

cantordust committed:
- Renamed `overwrite` to `bootstrap` in `Project.ensure_table()`.

cantordust committed:
- Simplified the entry points in `segment_images` (the decorated function for MaskFormer was moved to `models/maskformer/cli.py`).

cantordust committed:
- Removed `ray_actor_options` from the decoration in `serve/server.py` as it was causing errors. The proper configuration for deployments should be examined in a separate issue.

cantordust committed:
- Brief tutorial of the CLI (work in progress).

cantordust committed:
- Brief tutorial of the CLI (work in progress).

cantordust committed:
- Updated the `Project` class to create the `mapillary` table if it's missing (otherwise the CLI produces an error). The schema is somewhat arbitrary (all the fields are of type `str`, except `geometry`, which is of type `polygon`), so that needs to be updated to a proper schema corresponding to the Mapillary data types.

cantordust committed:
- Updated the `Project` class to create the `mapillary` table if it's missing (otherwise the CLI produces an error). The schema is somewhat arbitrary (all the fields are of type `str`, except `geometry`, which is of type `polygon`), so that needs to be updated to a proper schema corresponding to the Mapillary data types.

cantordust committed:
- Fixed nsightly file names for segmentation archives.
- Fixed BFMS segmentations not being saved in compressed format.

cantordust committed:
Merge branch 'cli_tutorial' of github.com:Urban-M4/streetscapes into cli_tutorial

cantordust committed:
- Small update to the database interfaces for BFMS and MaskFormer

cantordust committed:
Merge pull request #88 from Urban-M4/cli_tutorial

The beginnings of a CLI tutorial

cantordust committed:
- Updated the CLI definition

cantordust committed:
- Small updates to the BFMS servable model

cantordust committed:
- Small updates to the MaskFormer servable model

cantordust committed:
- Removed the deprecated `ModelBase` class

cantordust committed:
- Updated the `utils.functions` module:
	- Updated the label flattening function, it is now a prompt processing function.
	- Added a small function to get a torch.device object.

cantordust committed:
- Added a servable DinoSAM model.

cantordust committed:
- Added DinoSAM as an option to the CLI.

cantordust committed:
- Fixed label argument issue for MaskFormer.

cantordust committed:
- Small fix for the MaskFormer request schema.
